core/src/main/java/com/linecorp/armeria/common/multipart/Multipart.java

```diff
@@ -0,0 +1,288 @@
+     * > byte[] fileData = ...;
+     * > BodyPart filePart = BodyPart.builder()
+     * >                             .headers(headers)
+     * >                             .content(fileData)
+     * >                             .build();
+     *
+     * > RequestHeaders requestHeaders = RequestHeaders.of(HttpMethod.POST, "/upload");
+     * > HttpRequest request = Multipart.of(filePart).toHttpRequest(requestHeaders);
+     * > CompletableFuture<AggregatedHttpResponse> response = client.execute(request).aggregate();
+     * }</pre>
+     */
+    default HttpRequest toHttpRequest(RequestHeaders requestHeaders) {
+        requireNonNull(requestHeaders, "requestHeaders");
+        final String mediaTypeString = requestHeaders.get(HttpHeaderNames.CONTENT_TYPE);
+        final MediaType contentType;
+        if (mediaTypeString != null) {
+            contentType = MediaType.parse(mediaTypeString);
+            checkArgument("multipart".equals(contentType.type()),
+                          "Content-Type: %s (expected: multipart content type)", contentType);
+            contentType.withParameter("boundary", boundary());
+        } else {
+            contentType = MediaType.MULTIPART_FORM_DATA.withParameter("boundary", boundary());
+        }
+
+        final RequestHeaders updated = requestHeaders.withMutations(builder -> {
+            builder.add(HttpHeaderNames.CONTENT_TYPE, contentType.toString());
+        });
+        return HttpRequest.of(updated, this);
+    }
+
+    /**
+     * Returns the boundary string.
+     */
+    String boundary();
+
+    /**
+     * Returns all the nested body parts.
+     *
+     * <p>Note: Once a {@link BodyPart} is subscribed, you should subscribe to {@link BodyPart#content()}
+     * before subscribing to the next {@link BodyPart}.<pre>{@code
+     * > import reactor.core.publisher.Flux;
+     *
+     * > HttpRequest req = ...;
+     * > Multipart multiPart = Multipart.from(req);
+     *
+     * > // Good:
+     * > Flux.from(multiPart.bodyParts())
+     * >     .subscribe(bodyPart -> {
+     * >         Flux.from(bodyPart.content()) // Safely subscribe to BodyPart.content()
+     * >             .map(HttpData::toStringUtf8)
+     * >             .collectList()
+     * >             .subscribe(contents -> { ... });
+     * >     });
+     *
+     * > // Bad:
+     * > Flux.from(multiPart.bodyParts())
+     * >     .collectList() // This will subscribe BodyPart.content() first before you subscribe to it.
+     * >     .subscribe(bodyParts -> {
+     * >         bodyParts.forEach(part -> {
+     * >             Flux.from(part.content())
+     * >                 .collectList() // Throws IllegalStateException("Only single subscriber is allowed")
+     * >                 .subscribe(contents -> { ... });
+     * >         });
+     * >     });
+     * } </pre>
+     * If you don't know what this means, use {@link #aggregate()}.
+     */
+    Publisher<BodyPart> bodyParts();
```

Comment:
`@CheckReturnValue`?

Reply:
`StreamMessage<BodyPart>` just in case we implement this using our own `StreamMessage` implementations in the future.

Reply:
Now StreamMessage is returned.😀